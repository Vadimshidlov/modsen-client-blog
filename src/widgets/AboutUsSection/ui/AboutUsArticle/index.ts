export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
