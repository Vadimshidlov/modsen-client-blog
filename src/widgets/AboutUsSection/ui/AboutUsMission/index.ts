export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
