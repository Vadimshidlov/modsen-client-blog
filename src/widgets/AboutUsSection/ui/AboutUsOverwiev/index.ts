export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
