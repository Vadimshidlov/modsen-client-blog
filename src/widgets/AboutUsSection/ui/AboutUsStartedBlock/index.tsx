export { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock/AboutUsStartedBlock';
