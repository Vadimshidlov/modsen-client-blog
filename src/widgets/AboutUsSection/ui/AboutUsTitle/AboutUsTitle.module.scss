.aboutUsTitleContainer {
  position: relative;
  display: flex;
  justify-content: space-between;
  padding-right: 40px;
  padding-left: 9%;

  @media (max-width: 1100px) {
    flex-direction: column;
    padding-right: 20px;
    padding-left: 20px;
  }

  .leftSideWrapper {
    position: relative;
    z-index: 1;
    margin-bottom: -42px;
    padding: 64px;
    padding-top: 144px;
    background: $white;

    @media (max-width: 1100px) {
      padding: 32px;
    }

    .aboutUsSubTitle {
      font-size: $fz16;
      letter-spacing: 3px;
      margin-bottom: 16px;
    }

    .aboutUsTitle {
      max-width: 496px;
      @include headingOne;

      @media (max-width: 1300px) {
        @include headingTwo;
      }

      @media (max-width: 1100px) {
        max-width: 100%;
      }

      @media (max-width: $tablet) {
        @include headingThree;
      }

      @media (max-width: $mobile) {
        @include headingFive;
      }
    }
  }

  .aboutUsText {
    padding-top: 182px;
    padding-bottom: 80px;
    justify-self: end;
    max-width: 477px;
    @include bodyOne;

    @media (max-width: 1100px) {
      max-width: 100%;
      padding-top: 42px;
      padding-bottom: 20px;
    }
  }
}
