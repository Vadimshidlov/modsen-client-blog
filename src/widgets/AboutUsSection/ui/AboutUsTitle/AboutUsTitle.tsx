import React from 'react';
import styles from '@/widgets/AboutUsSection/ui/AboutUsTitle/AboutUsTitle.module.scss';
import { Inter } from 'next/font/google';

const inter = Inter({
  subsets: ['latin'],
  weight: '600',
  variable: '--font-inter',
});

export function AboutUsTitle() {
  return (
    <section className={styles.aboutUsTitleContainer}>
      <div className={styles.leftSideWrapper}>
        <h3 className={`${inter.variable} ${styles.aboutUsSubTitle}`}>ABOUT US</h3>
        <h1 className={styles.aboutUsTitle}>
          We are a team of content writers who share their learnings
        </h1>
      </div>
      <span className={styles.aboutUsText}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
        laboris nisi ut aliquip ex ea commodo consequat.
      </span>
    </section>
  );
}
