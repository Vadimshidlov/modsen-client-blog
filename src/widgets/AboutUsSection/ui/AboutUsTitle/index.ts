export { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle/AboutUsTitle';
