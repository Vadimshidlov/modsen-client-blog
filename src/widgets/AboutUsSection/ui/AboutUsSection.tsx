import React from 'react';
import styles from '@/widgets/AboutUsSection/ui/AboutUsSection.module.scss';
import container from '@/shared/styles/container.module.scss';
import { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle';
import { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev';
import { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission';
import { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam';
import { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock';

export function AboutUsSection() {
  return (
    <section className={`${styles.aboutUsSection} ${container.container}`}>
      <AboutUsTitle />
      <AboutUsOverwiev />
      <AboutUsMission />
      <AboutUsTeam />
      <AboutUsStartedBlock />
    </section>
  );
}
