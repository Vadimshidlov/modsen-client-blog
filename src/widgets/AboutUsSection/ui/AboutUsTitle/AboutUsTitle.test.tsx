import { render } from '@testing-library/react';
import { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle';

jest.mock('next/font/google', () => ({
  Inter: () => ({ variable: 'fake-variable' }),
}));

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('AboutUsTitle tests', () => {
  test('renders title and subtitle with correct text content', () => {
    const { getByText } = render(<AboutUsTitle />);

    expect(getByText('Mocked subTitle')).toBeInTheDocument();
    expect(getByText('Mocked title')).toBeInTheDocument();
  });

  test('should matches snapshot', () => {
    const { container } = render(<AboutUsTitle />);
    expect(container).toMatchSnapshot();
  });
});
