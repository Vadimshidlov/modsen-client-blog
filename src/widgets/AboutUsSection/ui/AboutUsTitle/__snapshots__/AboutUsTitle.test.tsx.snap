// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsTitle tests should matches snapshot 1`] = `
<div>
  <section
    class="aboutUsTitleContainer"
  >
    <div
      class="leftSideWrapper"
    >
      <h3
        class="fake-variable aboutUsSubTitle"
      >
        Mocked subTitle
      </h3>
      <h1
        class="aboutUsTitle"
      >
        Mocked title
      </h1>
    </div>
    <span
      class="aboutUsText"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </span>
  </section>
</div>
`;
